src/main/clojure/clara/rules/compiler.clj

```diff
@@ -378,10 +378,16 @@
              ~'?__bindings__ (atom ~initial-bindings)]
          ~(compile-constraints constraints)))))
 
+(defn build-token-assingment
+  "A helper function to build variable assignment forms for tokens."
+  [binding-key]
+  (list (symbol (name binding-key))
```

Comment:
You could experiment with accessing the :bindings by the literal field as well if this is a performance hotspot.  I don't know if it will matter in any siginificant way, but the keyword lookup does have overhead versus a Java field member access at least on the JVM.  I think (not sure offhand) the syntax for that to work in both clj and cljs is .-fieldname.

Reply:
Note: not needed for merge, just an observation/food for thought.